### PR TITLE
[LPS-122956,LPS-122966] Removes Server-side Closure Templates (Soy) Support

### PR DESCRIPTION
**Pull request by jbalsas**
This PR effectively **removes the possibility of rendering Soy templates from the server**.

To that effect, this PR:
- [**Updates** `portal-template-soy-renderer-impl` to avoid dependency with `soy-api` and allow a final upgrade path by allowing soy components to render basic placeholders before hydrating them
](https://github.com/liferay-frontend/liferay-portal/commit/d3d7ec6af3eed2e692897a9bf33c23b1a9950ca7)
- [**Removes** `portal-template-soy-api`, `portal-template-soy-impl` and `portal-template-soy-context-contributor`](https://github.com/liferay-frontend/liferay-portal/commit/0997027285ed6c281568e3b14ac36b9809654d96)
- [**Updates** `clay:management-toolbar` implementation to reduce layout shift when the component initializes](https://github.com/liferay-frontend/liferay-portal/commit/c0b3e63b9eac078e561f8e8512878c737ef33b95)
- [**Deprecates** remaining APIs](https://github.com/liferay-frontend/liferay-portal/commit/12068aa47916520cdd22747dc314a129d923a7fa)

A complete removal is not practical at this point because:
- The heavily used `clay:management-toolbar-v2` is still soy-based
- Commerce has still quite some components either extending `ComponentRendererTag` or using `ComponentRenderer` directly.

As explained in the commits, this PR takes a path in between by simply removing the server-side rendering path but maintaining the JS component initialization as can be appreciated in the following screencast:

mb

There are 2 main quirks that we can adjust in the future:
- Initial empty placeholder. While minimal, the placeholder is still very much an empty space that can't be linked to a toolbar. Adding some more elements like the add button or selection checkbox should help making the mental connection faster. Since we're working on the new version of the tag, I'd leave this for consideration there.
- An empty management toolbar is shorter than one with buttons, causing some visible movement in some transitions. This could also be solved in various ways, the simplest of which would be to set a consistent `min-height`.

**Test Plan**
- I've manually tested this lightly using `Documents & Media` and `Journal` apps and verifying the ManagementToolbar component renders and works as expected
- Run tests in CI. I'm particularly afraid of of race conditions now that the toolbar markup is not present until JS kicks in and CI is known to be slow to execute it.

**Future Plan**
- Communicate thoroughly that the rest of Soy support could still go away even if present in 7.4 GA1. This is up to @dsanz to figure out with the outreach team.
- Remove last usages (clay & commerce)
- Decide if we want to keep the modules around for an easier upgrade path (@dsanz, @david-truong?)

### CODE REVIEW

File: readme/BREAKING_CHANGES.markdown [hunk 1]
```diff
@@ -381,4 +381,62 @@ away from soy:
...
+#### What changed?
+
+The following modules and the classes they exported to allow Soy rendering
+server-side have been removed:
+- `portal-template-soy-api`
+- `portal-template-soy-impl`
+- `portal-template-soy-context-contributor`
+
+To simplify the migration, the following modules remain available in a deprecated
+deprecated fashion providing only client-side initialization of previous soy
```
Comment (wincent) [inline readme/BREAKING_CHANGES.markdown:399]: "deprecated deprecated"

also: should always capitalize the "s" in "soy"

**Review by wincent — COMMENTED** [review]
I'm not qualified to review this, but changes look good to me.

**Comment by jbalsas** [comment]
/cc @liferay-commerce, @liferay-forms, as far as I can tell, you have the following Soy usages. Could you double check to verify nothing is severely broken with this? After it, we should plan next steps to get rid of those 😉 

#### Forms
- Direct usage of `ComponentRenderer`
    - [DDMFormRendererImpl](https://github.com/liferay-frontend/liferay-portal/blob/master/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/java/com/liferay/dynamic/data/mapping/form/renderer/internal/DDMFormRendererImpl.java#L94)

#### Commerce
- Direct usage of `ComponentRenderer`
    - [MiniumCPContentListEntryRenderer](https://github.com/liferay-frontend/liferay-portal/blob/master/modules/apps/commerce/commerce-theme-minium/commerce-theme-minium-impl/src/main/java/com/liferay/commerce/theme/minium/internal/product/renderer/list/entry/MiniumCPContentListEntryRenderer.java#L333)
    - [MiniumCPCompareListRenderer](https://github.com/liferay-frontend/liferay-portal/blob/master/modules/apps/commerce/commerce-theme-minium/commerce-theme-minium-impl/src/main/java/com/liferay/commerce/theme/minium/internal/product/renderer/list/MiniumCPCompareListRenderer.java#L182)
- Extending `ComponentRendererTag`
    - [AddressModalTag](https://github.com/liferay-frontend/liferay-portal/blob/master/modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/AddressModalTag.java#L35)
    - [AddToCartTag](https://github.com/liferay-frontend/liferay-portal/blob/master/modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/AddToCartTag.java#L42)
    - [CompareCheckboxTag](https://github.com/liferay-frontend/liferay-portal/blob/master/modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/CompareCheckboxTag.java#L40)
    - [PriceTag](https://github.com/liferay-frontend/liferay-portal/blob/master/modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/PriceTag.java#L48)
    - [ProductDetailsModalTag](https://github.com/liferay-frontend/liferay-portal/blob/master/modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/ProductDetailsModalTag.java#L27)
    - [QuantitySelectorTag](https://github.com/liferay-frontend/liferay-portal/blob/master/modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/QuantitySelectorTag.java#L25)
    - [SearchBarTag](https://github.com/liferay-frontend/liferay-portal/blob/master/modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/SearchBarTag.java#L32)
    - [SearchResultsTag.java](https://github.com/liferay-frontend/liferay-portal/blob/master/modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/SearchResultsTag.java#L34)
    - [UserInvitationModalTag](https://github.com/liferay-frontend/liferay-portal/blob/master/modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/UserInvitationModalTag.java#L28)
    - [UserRolesModalTag](https://github.com/liferay-frontend/liferay-portal/blob/master/modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/UserRolesModalTag.java#L44)

File: readme/BREAKING_CHANGES.markdown (hunk 1 above)
Comment (jbalsas) [reply readme/BREAKING_CHANGES.markdown:399]: > also: should always capitalize the "s" in "soy"

You won't believe it, but I promise I double-checked all of them 😭

**Comment by jbalsas** [comment]
This should actually break Forms Email notifications until [LPS-124398 Use ftl template instead of soy](https://github.com/natocesarrego/liferay-portal/pull/938) is merged. Hopefully we can get both in today! 🤞 

/cc @carolmariaabb, @natocesarrego

**Comment by jbalsas** [comment]
ci:test:sf

**Comment by jbalsas** [comment]
Thanks to @gianmarcobrunialti for kindly validating this from commerce side:

> @Chema Balsas Checked out your PR / ant all => everything works in Commerce front store (which is the main reason why those Soy dependencies still exist :D)
>
> I was expecting nothing other than a safety-first approach, @Chema Balsas thank ya for the smoothness 

shipit

**Comment by jbalsas** [comment]
ci:test:relevant

**Comment by jbalsas** [comment]
@john-co, could you please prioritize analyzing this one?

**Comment by john-co** [comment]
ci:test:bundle

**Comment by john-co** [comment]
Found 3 main unique failures that seem related:

[Related]
- [ContentPages#ViewSelectedImageName](https://testray.liferay.com/home/-/testray/cases/1310738164)
- [ContentPagesWithEditables#AddContentPageWithImageFragment](https://testray.liferay.com/reports/production/logs/test-1-6/1607969081352/test-portal-acceptance-pullrequest(master)/7665/functional-tomcat90-mysql57-jdk8/2/10/LocalFile.ContentPagesWithEditables_AddContentPageWithImageFragment/index.html.gz)
`LIFERAY_ERROR: Unable to save metadata for file version 57656`
image
`Element is not present at "//div[contains(@class,'management-bar')]//span[contains(@class,'navbar-text')]"`
image
- [OpenGraphPreview#Smoke](https://test-1-13.liferay.com/job/test-portal-acceptance-pullrequest-batch(master)/AXIS_VARIABLE=6,label_exp=!master/419933//testReport/com.liferay.poshi.runner/PoshiRunner/test_LocalFile_OpenGraphPreview_Smoke_)
image
```
unknown error: unhandled inspector error: {"code":-32000,"message":"Cannot find context with specified id"}
  (Session info: chrome=86.0.4240.111)
  (Driver info: chromedriver=2.37.544315 (730aa6a5fdba159ac9f4c1e8cbc59bf1b5ce12b7),platform=Linux 4.14.138-rancher x86_64) (WARNING: The server did not provide any stacktrace information)
Command duration or timeout: 0 milliseconds
Build info: version: '3.141.59', revision: 'e82be7d358', time: '2018-11-14T08:17:03'
System info: host: 'cloud-10-0-110-228.lax.liferay.com', ip: '127.0.0.1', os.name: 'Linux', os.arch: 'amd64', os.version: '4.14.138-rancher', java.version: '1.8.0_121'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Capabilities {acceptInsecureCerts: false, acceptSslCerts: false, applicationCacheEnabled: false, browserConnectionEnabled: false, browserName: chrome, chrome: {chromedriverVersion: 2.37.544315 (730aa6a5fdba15..., userDataDir: /tmp/.org.chromium.Chromium...}, cssSelectorsEnabled: true, databaseEnabled: false, handlesAlerts: true, hasTouchScreen: false, javascriptEnabled: true, locationContextEnabled: true, mobileEmulationEnabled: false, nativeEvents: true, networkConnectionEnabled: false, pageLoadStrategy: normal, platform: LINUX, platformName: LINUX, rotatable: false, setWindowRect: true, takesHeapSnapshot: true, takesScreenshot: true, unexpectedAlertBehaviour: , unhandledPromptBehavior: , version: 86.0.4240.111, webStorageEnabled: true}
Session ID: 3ec5ffe70ed16b6563bf9461806ac369
```

I'll need to dig some more manually with the test bundle.

**Comment by john-co** [comment]
For ContentPages#ViewSelectedImageName:

It looks like there's now a search icon that wasn't there before. The search icon also overlaps the Add Button. I believe the test tries to click on the Add Button but because the search icon overlaps on top of the Add Button the test clicks the search icon and brings us back to the selection of images.
SearchIconOverlapWithAddButton


Here is what we expect prior to the commit.
ExpectedImageUploadModalClickAdd

**Comment by jbalsas** [comment]
Thanks @john-co! The problem of the search icon overlapping is secondary. It seems to be caused by the toolbar replacing the navigation rather than instantiating on its own. I'll analyze why that's the case.

**Comment by jbalsas** [comment]
ci:test:sf

**Comment by jbalsas** [comment]
ci:test:relevant

**Comment by jbalsas** [comment]
ci:test:sf

**Comment by jbalsas** [comment]
ci:forward

**Comment by jbalsas** [comment]
Hey @john-co, could you take a look at these failured again? Are these still related?

**Comment by john-co** [comment]
ci:test:bundle

**Comment by john-co** [comment]
[Related]
- CPDynamicdatalists#DeleteList - When creating a new DDL list, clicking on data definition link does nothing.
DataDefinitionsNotClickable
- CPCommerce - looks like the xpath doesn't work anymore on the commerce account selector. If we're intentionally removing the `data-onclick="toggleAccountSelector"` class then need a test fix to update test. Browser console errors look unrelated since they appear in upstream too. cc/ @iovaneale 
CommerceDropdown 
here's the expected html:
screenshot_15


[Unrelated]
- Forms#ViewFieldValidationIsNotBypassedWhenNextButtonIsHitTwice - rare flaky
https://liferay.spiraservice.net/16/TestCase/202212.aspx

[Not sure yet]
- CPSitetemplates#DeleteSiteTemplate - looks new...it failed in the last [two test runs](https://liferay.spiraservice.net/16/TestCase/191769.aspx) but I can't reproduce issue locally...possibly ci lag? cc/ @david-gutierrez-mesa

**Comment by jbalsas** [comment]
Thanks for the help @john-co!!

- Found the issue in CPDynamicdatalists#DeleteList. Trying again.
- The commerce thing should be a test fix. I have a fair idea of why that attribute was there to begin with, but it was not really expected to be there so the current behaviour is correct. I've added a commit with a crude attempt at it to see if we can get this through, but maybe @iovaneale or the team want to adjust it to something different later.
- CPSitetemplates#DeleteSiteTemplate waiting on @david-gutierrez-mesa to weigh in.

**Comment by jbalsas** [comment]
ci:test:sf

**Comment by jbalsas** [comment]
>      [java] java.lang.Exception: Found 1 formatting issues:
>     [java] 1: Do not use arrow function, see https://github.com/liferay/liferay-portal/blob/master/modules/util/source-formatter/src/main/resources/documentation/checks/jsp_arrow_function_check.markdown: ./modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/select_structure.jsp 111 (SourceCheck:JSPArrowFunctionCheck)

@wincent, do we have an issue to relax jsp checks for `7.4` akin to what we do for `7.3` so we can write arrow functions in jsps? Or was there a reason why we still wouldn't want to do it?

**Comment by wincent** [comment]
> > ```
> > [java] java.lang.Exception: Found 1 formatting issues:
> > [java] 1: Do not use arrow function, see https://github.com/liferay/liferay-portal/blob/master/modules/util/source-formatter/src/main/resources/documentation/checks/jsp_arrow_function_check.markdown: ./modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/select_structure.jsp 111 (SourceCheck:JSPArrowFunctionCheck)
> > ```
> 
> @wincent, do we have an issue to relax jsp checks for `7.4` akin to what we do for `7.3` so we can write arrow functions in jsps? Or was there a reason why we still wouldn't want to do it?

There is no reason not to do it (except for code that might be backported, obviously).

Let me see if I can dig up a reference... [We are at least ready for it on our side](https://github.com/liferay/liferay-frontend-projects/pull/214/commits/72cbe61e5e503e202e4bfa85eca4b2267f668eb4), but I can't find anything else in my email so we probably never did anything else about it. I imagine the reason why is to avoid _proactively_ churning code that might make backporting harder, but there is certainly nothing wrong with _allowing_ it.

**Comment by jbalsas** [comment]
> There is no reason not to do it (except for code that might be backported, obviously).

That might be reason enough... let's keep it as is and hope we don't touch JSPs that much in `7.4`

**Comment by jbalsas** [comment]
ci:test:sf

**Comment by jbalsas** [comment]
ci:test:relevant

**Comment by jbalsas** [comment]
> LocalFile.CPSitetemplates#DeleteSiteTemplate

Also related... hopefully the last one. Will push again 🤞

**Comment by jbalsas** [comment]
ci:test:sf

**Comment by jbalsas** [comment]
ci:test:relevant

**Comment by jbalsas** [comment]
> This pull contains no unique failures.

That seals it!

**Comment by jbalsas** [comment]
ci:forward

**Pull request closed without merging**